modules/runner/src/main/java/org/apache/ignite/internal/app/LifecycleManager.java

```diff
@@ -81,10 +84,13 @@ void startComponent(IgniteComponent component) throws NodeStoppingException {
      * @param components Ignite components to start.
      * @throws NodeStoppingException If node stopping intention was detected.
      */
-    void startComponents(IgniteComponent... components) throws NodeStoppingException {
+    CompletableFuture<Void> startComponents(IgniteComponent... components) throws NodeStoppingException {
+        List<CompletableFuture<Void>> startFutures = new ArrayList<>();
```

Comment:
This should be an array, not an `ArrayList`. This way we can avoid copying in the end. But I also think that components should be started sequentially and not in parallel, at least for now

Reply:
refactored